test/cfoa/visit_tests.cpp

```diff
@@ -971,6 +976,158 @@ namespace {
   boost::unordered::concurrent_flat_set<raii, transp_hash,
     transp_key_equal>* transp_set;
 
+  struct mutable_pair
+  {
+    mutable_pair(int first_ = 0, int second_ = 0):
+      first{first_}, second{second_} {}
+
+    int         first = 0;
+    mutable int second = 0;
+  };
+
+  struct null_mutable_pair
+  {
+    operator mutable_pair() const { return {0,0}; }
+  };
+
+  struct mutable_pair_hash
+  {
+    using is_transparent = void;
+
+    std::size_t operator()(const mutable_pair& x) const
+    {
+      return boost::hash<int>{}(x.first);
+    }
+
+    std::size_t operator()(const null_mutable_pair&) const
+    {
+      return boost::hash<int>{}(0);
+    }
+  };
+
+  struct mutable_pair_equal_to
+  {
+    using is_transparent = void;
+
+    bool operator()(const mutable_pair& x, const mutable_pair& y) const
+    {
+      return x.first == y.first;
+    }
+
+    bool operator()(const null_mutable_pair&, const mutable_pair& y) const
+    {
+      return 0 == y.first;
+    }
+
+    bool operator()(const mutable_pair& x, const null_mutable_pair&) const
+    {
+      return x.first == 0;
+    }
+  };
+
+  template<typename F>
+  void exclusive_access_for(F f)
+  {
+    std::atomic_int             num_started{0};
+    std::atomic_int             in_visit{0};
+    std::mutex                  m;
+    std::condition_variable     cv;
+    bool                        finish = false;
+
+    auto bound_f = [&] {
+      ++num_started;
+      f([&] (const mutable_pair& x) {
+        ++in_visit;
+        ++x.second;
+
+        {
+          std::unique_lock<std::mutex> lk(m);
+          cv.wait(lk, [&] { return finish; });
```

Comment:
I think we should have a latch component either internally or in Compat we can use here. I'm assuming the intention here was to block all the spawned threads until we're ready for them to actually run, right?

Reply:
You're absolutely right! Changed it to use `boost::compat::latch`.